### ISSUE TITLE
Error invalid_token - The audience is invalid

### ISSUE THREAD

**Issue by gwenael-louandre** [Azure-Samples/ms-identity-javascript-angular-tutorial#8]
# Issue

The front authentication is well but when I request the backend I have a 401 response with : 

www-authenticate | Bearer error="invalid_token", error_description="The audience 'xxxxxxx' is invalid"
-- | --

## This issue is for the sample

```console
    - [ ] 1-1) Sign-in with Azure AD
    - [ ] 1-2) Sign-in with Azure AD B2C
    - [ ] 2-1) Acquire a Token and call Microsoft Graph
    - [x ] 3-1) Protect and call a web API on Azure AD
    - [ ] 3-2) Protect and call a web API on Azure AD B2C
    - [ ]   4) Deploy to Azure Storage and App Service
    - [ ] 5-1) Call a web API using App Roles
    - [ ] 5-2) Call a web API using Security Groups
    - [ ] 6-1) Call a multi-tenant web API
    - [ ] 7-1) Call Microsoft Graph using on-behalf-of flow
    - [ ] 7-2) Call a web API using Proof of Possession tokens
```

## This issue is for a

```console
    - [ ] bug report -> please search issues before submitting
    - [ x] question
    - [ ] feature request
    - [ ] documentation issue or request
```

### Minimal steps to reproduce

>I just followed the steps. The only modification is that I updated azure/msal-angular (2.0.0-beta.6) and @azure/msal-browser (2.14.2)

### Any log messages given by the failure

>Here is my token
{
  "aud": "xxxxxxxxx",
  "iss": "https://login.microsoftonline.com/98731832-xxxxxxxx-f68390800051/v2.0",
  "iat": 1622482003,
  "nbf": 1622482003,
  "exp": 1622485903,
  "aio": "AVQAq/8TAAAApcE99DC5d8+fiGXeX+4YQiGQk2LRAkoW0/2ef6TjiO9eFSxeWuF4+GL4awP6FUtCGFN89WltByAUrmjhuOASSNT9TXUU+wzhGN0siQXU5pc=",
  "azp": "c8919a3e-aaaa-42c0-b573-533e717bacb8",
  "azpacr": "0",
  "name": "LOUANDRE Gwenael",
  "oid": "d7bbb0e4-2dd3-4c76-9160-uuuuuu",
  "preferred_username": "gwenael.louandre@vinci-construction.com",
  "rh": "0.AQUAMhhzmNpARkWSA_aDkIAAUT6akciqqsBCtXNTPnF7rLgFAHY.",
  "scp": "access_as_user",
  "sub": "BomEbk_RNwYiVRKpjPfANmUhtFzkj3kCef1qMH0iatU",
  "tid": "98731832-40da-4546-9203-f68390800051",
  "uti": "MqVE0aYsTEautUNWTdMhAQ",
  "ver": "2.0"
}

### Browser and version

> Chrome, Firefox

I found several posts on stackoverflow about that problem but no one corrects the problem. What am I doing wrong?
Thanks! We'll be in touch soon.

**Comment by derisen** [comment]
@gwenael-louandre can you share your configuration file for the SPA and API? Also any logs in the browser console would be useful. 

The audience in the token should be the same with the Client Id of your backend. Basically these values should be equal:

- `aud` claim in your token (does it look like `api://xxxxx` or just `xxxxx`)
- `ClientId` in _API/appsettings.json_
- `protectedResources.todoListApi.scopes` in _SPA/src/app/auth-config.ts_ (the part before the scope name)


Btw, `msal-angular` is out of beta now -please update your package to the latest.

**Comment by gwenael-louandre** [comment]
Hello,
Thanks for your answer.

Here is my auth-config.ts : 
![image](https://user-images.githubusercontent.com/4600171/120324558-0f9b4500-c2e7-11eb-83b5-c4b24c4e939d.png)

Here is the appsettings.json : 
![image](https://user-images.githubusercontent.com/4600171/120324727-3c4f5c80-c2e7-11eb-8554-66d76409b303.png)

Here is the Azure SPA configuration : 
![image](https://user-images.githubusercontent.com/4600171/120325266-c8618400-c2e7-11eb-8fda-3aa078b8949e.png)

![image](https://user-images.githubusercontent.com/4600171/120325370-dd3e1780-c2e7-11eb-82ee-d8343f4aa67e.png)

![image](https://user-images.githubusercontent.com/4600171/120325527-08c10200-c2e8-11eb-948f-3bb7330891b8.png)

Here is the Azure API configuration : 
![image](https://user-images.githubusercontent.com/4600171/120325770-46be2600-c2e8-11eb-90e7-0f6b54ad670a.png)
![image](https://user-images.githubusercontent.com/4600171/120325909-6ce3c600-c2e8-11eb-80a4-9ced944701b0.png)

The aud claim in token is just xxxxx : 
![image](https://user-images.githubusercontent.com/4600171/120326272-d8c62e80-c2e8-11eb-8aa4-96dceccbd9eb.png)

So about your question, the three points seem to be the same.
[localhost-1622552382083.log](https://github.com/Azure-Samples/ms-identity-javascript-angular-tutorial/files/6576393/localhost-1622552382083.log)

I attached the log from the browser.

I just checked the release (https://github.com/AzureAD/microsoft-authentication-library-for-js/releases) and the last version I see is @azure/msal-angular v2.0.0-beta.6 What is the last version?

**Comment by derisen** [comment]
@gwenael-louandre it looks like `ClientId` in `API/appsettings.json` doesn't match the Application (client) Id on Azure portal (screenshot for Azure API configuration). Can you double check that?

**Comment by gwenael-louandre** [comment]
Thanks! That was my problem. I mixed two projects I worked at the same time. 
On the other hand, I have a question about one step in demo. In the Register the client app (msal-angular-spa) paragraph after creating the client app, I added a single page application platform in the 'Authentication' Azure menu. I think if we don't declare this kind of platform the msal-angular doesn't work. Am I wrong?
